x-pack/plugins/cloud/public/plugin.ts

```diff
@@ -82,19 +88,39 @@ export class CloudPlugin implements Plugin<CloudSetup> {
   public start(coreStart: CoreStart, { security }: CloudStartDependencies) {
     const { deployment_url: deploymentUrl, base_url: baseUrl } = this.config;
     coreStart.chrome.setHelpSupportUrl(ELASTIC_SUPPORT_LINK);
-    if (baseUrl && deploymentUrl) {
-      coreStart.chrome.setCustomNavLink({
-        title: i18n.translate('xpack.cloud.deploymentLinkLabel', {
-          defaultMessage: 'Manage this deployment',
-        }),
-        euiIconType: 'arrowLeft',
-        href: getFullCloudUrl(baseUrl, deploymentUrl),
-      });
-    }
 
-    if (security && this.isCloudEnabled) {
-      const userMenuLinks = createUserMenuLinks(this.config);
-      security.navControlService.addUserMenuLinks(userMenuLinks);
-    }
+    this.checkIfAuthorizedForLinks().then((authorized) => {
+      if (authorized && baseUrl && deploymentUrl) {
+        coreStart.chrome.setCustomNavLink({
+          title: i18n.translate('xpack.cloud.deploymentLinkLabel', {
```

Comment:
NIT: I know `checkIfAuthorizedForLinks` cannot likely throw, but we may still want to add a `catch` handler, just to follow the best practices against UnhandledPromiseRejection, wdyt?

Reply:
++

Reply:
Good idea, will do!